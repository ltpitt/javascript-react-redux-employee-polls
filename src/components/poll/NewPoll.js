import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { useState } from "react";
import { addQuestion } from "../../actions/questions";
import Nav from "../nav/Nav";

const NewPoll = ({ dispatch }) => {
  const navigate = useNavigate();
  const [firstOption, setFirstOption] = useState("");
  const [secondOption, setSecondOption] = useState("");

  const handleFirstOptionChange = (e) => {
    setFirstOption(e.target.value);
  };

  const handleSecondOptionChange = (e) => {
    setSecondOption(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(addQuestion(firstOption, secondOption));
    navigate("/");
  };

  return (
    <div>
      <Nav />
      <div className="container">
        <h1>New Poll</h1>
        <form onSubmit={handleSubmit}>
          <div>
            <label data-testid="firstOptionLabel">First Option</label>
            <div className="mt-1">
              <input
                value={firstOption}
                onChange={handleFirstOptionChange}
                type="text"
                name="firstOption"
                id="firstOption"
                data-testid="firstOption"
              />
            </div>
          </div>

          <div>
            <label data-testid="secondOptionLabel">Second Option</label>
            <div>
              <input
                value={secondOption}
                onChange={handleSecondOptionChange}
                type="text"
                name="secondOption"
                id="secondOption"
                data-testid="secondOption"
              />
            </div>
          </div>

          <div>
            <button type="submit" data-testid="submit-poll">
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default connect()(NewPoll);
