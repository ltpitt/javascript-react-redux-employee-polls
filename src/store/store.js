import { configureStore } from "@reduxjs/toolkit";
import authedUserReducer from "../reducers/authedUser";
import questionsReducer from "../reducers/questions";
import usersReducer from "../reducers/users";
import logger from "../middleware/logger";
import thunk from "redux-thunk";
import { combineReducers } from "@reduxjs/toolkit";

export const store = configureStore({
  reducer: combineReducers({
    authedUser: authedUserReducer,
    questions: questionsReducer,
    users: usersReducer,
  }),
  middleware: [thunk, logger],
});
