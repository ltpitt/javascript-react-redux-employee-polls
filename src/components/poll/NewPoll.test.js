import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../store/store";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import NewPoll from "./NewPoll";

describe("NewPoll", () => {
  it("should render the NewPoll", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPoll />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("should render and match snapshot", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPoll />
        </BrowserRouter>
      </Provider>
    );

    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("should display all elements", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPoll />
        </BrowserRouter>
      </Provider>
    );

    const firstOptionInputElement = component.getByTestId("firstOption");
    const secondOptionInputElement = component.getByTestId("secondOption");
    const firstOptionLabelElement = component.getByTestId("firstOptionLabel");
    const secondOptionLabelElement = component.getByTestId("secondOptionLabel");
    const submitButtonElement = component.getByTestId("submit-poll");

    expect(firstOptionInputElement).toBeTruthy();
    expect(secondOptionInputElement).toBeTruthy();
    expect(firstOptionLabelElement).toBeTruthy();
    expect(secondOptionLabelElement).toBeTruthy();
    expect(submitButtonElement).toBeTruthy();
  });

  it("should allow to type first option", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPoll />
        </BrowserRouter>
      </Provider>
    );

    const firstOptionInputElement = component.getByTestId("firstOption");
    expect(firstOptionInputElement).toHaveValue("");

    fireEvent.change(firstOptionInputElement, { target: { value: "Meters" } });
    expect(firstOptionInputElement.value).toBe("Meters");
  });

  it("should allow to type second option", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPoll />
        </BrowserRouter>
      </Provider>
    );

    const secondOptionInputElement = component.getByTestId("secondOption");
    expect(secondOptionInputElement).toHaveValue("");

    fireEvent.change(secondOptionInputElement, { target: { value: "Feet" } });
    expect(secondOptionInputElement.value).toBe("Feet");
  });
});
