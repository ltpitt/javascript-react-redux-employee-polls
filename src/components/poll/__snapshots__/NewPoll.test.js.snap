// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll should render and display all elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <nav
            class="navbar navbar-expand-lg navbar-light bg-light"
          >
            <div
              class="container-fluid"
            >
              <a
                class="nav-link"
                href="/"
              >
                <img
                  alt=""
                  height="30"
                  loading="lazy"
                  src="/logo192.png"
                />
              </a>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                class="navbar-toggler"
                data-bs-target="#navbarSupportedContent"
                data-bs-toggle="collapse"
                type="button"
              >
                <i
                  class="fas fa-bars"
                />
              </button>
              <div
                class="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  class="navbar-nav me-auto mb-2 mb-lg-0"
                >
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/leaderboard"
                    >
                      Leaderboard
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/add"
                    >
                      New
                    </a>
                  </li>
                </ul>
                <div
                  data-testid="loggedUserName"
                  id="loggedUserName"
                >
                  Login -&gt;
                </div>
                <ul
                  class="navbar-nav mb-2 mb-lg-0"
                >
                  <li
                    class="nav-item dropdown"
                  >
                    <a
                      aria-expanded="false"
                      class="nav-link dropdown-toggle"
                      data-bs-toggle="dropdown"
                      data-testid="navbarDropdown"
                      href="#"
                      id="navbarDropdown"
                      role="button"
                    >
                      <img
                        class="rounded-circle img-fluid"
                        height="25"
                        id="authedUserAvatar"
                        src="/login192.png"
                        width="25"
                      />
                    </a>
                    <ul
                      aria-labelledby="navbarDropdown"
                      class="dropdown-menu dropdown-menu-end p-1"
                    >
                      <li>
                        <hr
                          class="dropdown-divider"
                        />
                      </li>
                      <li>
                        <a
                          class="nav-link"
                          to="#"
                        >
                          Log Out
                        </a>
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
        </div>
        <div
          class="container"
        >
          <h1>
            New Poll
          </h1>
          <form>
            <div>
              <label
                data-testid="firstOptionLabel"
              >
                First Option
              </label>
              <div
                class="mt-1"
              >
                <input
                  data-testid="firstOption"
                  id="firstOption"
                  name="firstOption"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                data-testid="secondOptionLabel"
              >
                Second Option
              </label>
              <div>
                <input
                  data-testid="secondOption"
                  id="secondOption"
                  name="secondOption"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <button
                data-testid="submit-poll"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light"
        >
          <div
            class="container-fluid"
          >
            <a
              class="nav-link"
              href="/"
            >
              <img
                alt=""
                height="30"
                loading="lazy"
                src="/logo192.png"
              />
            </a>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-bs-target="#navbarSupportedContent"
              data-bs-toggle="collapse"
              type="button"
            >
              <i
                class="fas fa-bars"
              />
            </button>
            <div
              class="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                class="navbar-nav me-auto mb-2 mb-lg-0"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/leaderboard"
                  >
                    Leaderboard
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/add"
                  >
                    New
                  </a>
                </li>
              </ul>
              <div
                data-testid="loggedUserName"
                id="loggedUserName"
              >
                Login -&gt;
              </div>
              <ul
                class="navbar-nav mb-2 mb-lg-0"
              >
                <li
                  class="nav-item dropdown"
                >
                  <a
                    aria-expanded="false"
                    class="nav-link dropdown-toggle"
                    data-bs-toggle="dropdown"
                    data-testid="navbarDropdown"
                    href="#"
                    id="navbarDropdown"
                    role="button"
                  >
                    <img
                      class="rounded-circle img-fluid"
                      height="25"
                      id="authedUserAvatar"
                      src="/login192.png"
                      width="25"
                    />
                  </a>
                  <ul
                    aria-labelledby="navbarDropdown"
                    class="dropdown-menu dropdown-menu-end p-1"
                  >
                    <li>
                      <hr
                        class="dropdown-divider"
                      />
                    </li>
                    <li>
                      <a
                        class="nav-link"
                        to="#"
                      >
                        Log Out
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
      <div
        class="container"
      >
        <h1>
          New Poll
        </h1>
        <form>
          <div>
            <label
              data-testid="firstOptionLabel"
            >
              First Option
            </label>
            <div
              class="mt-1"
            >
              <input
                data-testid="firstOption"
                id="firstOption"
                name="firstOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              data-testid="secondOptionLabel"
            >
              Second Option
            </label>
            <div>
              <input
                data-testid="secondOption"
                id="secondOption"
                name="secondOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <button
              data-testid="submit-poll"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NewPoll should render and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <nav
            class="navbar navbar-expand-lg navbar-light bg-light"
          >
            <div
              class="container-fluid"
            >
              <a
                class="nav-link"
                href="/"
              >
                <img
                  alt=""
                  height="30"
                  loading="lazy"
                  src="/logo192.png"
                />
              </a>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                class="navbar-toggler"
                data-bs-target="#navbarSupportedContent"
                data-bs-toggle="collapse"
                type="button"
              >
                <i
                  class="fas fa-bars"
                />
              </button>
              <div
                class="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  class="navbar-nav me-auto mb-2 mb-lg-0"
                >
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/leaderboard"
                    >
                      Leaderboard
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/add"
                    >
                      New
                    </a>
                  </li>
                </ul>
                <div
                  data-testid="loggedUserName"
                  id="loggedUserName"
                >
                  Login -&gt;
                </div>
                <ul
                  class="navbar-nav mb-2 mb-lg-0"
                >
                  <li
                    class="nav-item dropdown"
                  >
                    <a
                      aria-expanded="false"
                      class="nav-link dropdown-toggle"
                      data-bs-toggle="dropdown"
                      data-testid="navbarDropdown"
                      href="#"
                      id="navbarDropdown"
                      role="button"
                    >
                      <img
                        class="rounded-circle img-fluid"
                        height="25"
                        id="authedUserAvatar"
                        src="/login192.png"
                        width="25"
                      />
                    </a>
                    <ul
                      aria-labelledby="navbarDropdown"
                      class="dropdown-menu dropdown-menu-end p-1"
                    >
                      <li>
                        <hr
                          class="dropdown-divider"
                        />
                      </li>
                      <li>
                        <a
                          class="nav-link"
                          to="#"
                        >
                          Log Out
                        </a>
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
        </div>
        <div
          class="container"
        >
          <h1>
            New Poll
          </h1>
          <form>
            <div>
              <label
                data-testid="firstOptionLabel"
              >
                First Option
              </label>
              <div
                class="mt-1"
              >
                <input
                  data-testid="firstOption"
                  id="firstOption"
                  name="firstOption"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                data-testid="secondOptionLabel"
              >
                Second Option
              </label>
              <div>
                <input
                  data-testid="secondOption"
                  id="secondOption"
                  name="secondOption"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <button
                data-testid="submit-poll"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light"
        >
          <div
            class="container-fluid"
          >
            <a
              class="nav-link"
              href="/"
            >
              <img
                alt=""
                height="30"
                loading="lazy"
                src="/logo192.png"
              />
            </a>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-bs-target="#navbarSupportedContent"
              data-bs-toggle="collapse"
              type="button"
            >
              <i
                class="fas fa-bars"
              />
            </button>
            <div
              class="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                class="navbar-nav me-auto mb-2 mb-lg-0"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/leaderboard"
                  >
                    Leaderboard
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/add"
                  >
                    New
                  </a>
                </li>
              </ul>
              <div
                data-testid="loggedUserName"
                id="loggedUserName"
              >
                Login -&gt;
              </div>
              <ul
                class="navbar-nav mb-2 mb-lg-0"
              >
                <li
                  class="nav-item dropdown"
                >
                  <a
                    aria-expanded="false"
                    class="nav-link dropdown-toggle"
                    data-bs-toggle="dropdown"
                    data-testid="navbarDropdown"
                    href="#"
                    id="navbarDropdown"
                    role="button"
                  >
                    <img
                      class="rounded-circle img-fluid"
                      height="25"
                      id="authedUserAvatar"
                      src="/login192.png"
                      width="25"
                    />
                  </a>
                  <ul
                    aria-labelledby="navbarDropdown"
                    class="dropdown-menu dropdown-menu-end p-1"
                  >
                    <li>
                      <hr
                        class="dropdown-divider"
                      />
                    </li>
                    <li>
                      <a
                        class="nav-link"
                        to="#"
                      >
                        Log Out
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
      <div
        class="container"
      >
        <h1>
          New Poll
        </h1>
        <form>
          <div>
            <label
              data-testid="firstOptionLabel"
            >
              First Option
            </label>
            <div
              class="mt-1"
            >
              <input
                data-testid="firstOption"
                id="firstOption"
                name="firstOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              data-testid="secondOptionLabel"
            >
              Second Option
            </label>
            <div>
              <input
                data-testid="secondOption"
                id="secondOption"
                name="secondOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <button
              data-testid="submit-poll"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NewPoll should render the NewPoll 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container"
        >
          <nav
            class="navbar navbar-expand-lg navbar-light bg-light"
          >
            <div
              class="container-fluid"
            >
              <a
                class="nav-link"
                href="/"
              >
                <img
                  alt=""
                  height="30"
                  loading="lazy"
                  src="/logo192.png"
                />
              </a>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                class="navbar-toggler"
                data-bs-target="#navbarSupportedContent"
                data-bs-toggle="collapse"
                type="button"
              >
                <i
                  class="fas fa-bars"
                />
              </button>
              <div
                class="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  class="navbar-nav me-auto mb-2 mb-lg-0"
                >
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/leaderboard"
                    >
                      Leaderboard
                    </a>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      href="/add"
                    >
                      New
                    </a>
                  </li>
                </ul>
                <div
                  data-testid="loggedUserName"
                  id="loggedUserName"
                >
                  Login -&gt;
                </div>
                <ul
                  class="navbar-nav mb-2 mb-lg-0"
                >
                  <li
                    class="nav-item dropdown"
                  >
                    <a
                      aria-expanded="false"
                      class="nav-link dropdown-toggle"
                      data-bs-toggle="dropdown"
                      data-testid="navbarDropdown"
                      href="#"
                      id="navbarDropdown"
                      role="button"
                    >
                      <img
                        class="rounded-circle img-fluid"
                        height="25"
                        id="authedUserAvatar"
                        src="/login192.png"
                        width="25"
                      />
                    </a>
                    <ul
                      aria-labelledby="navbarDropdown"
                      class="dropdown-menu dropdown-menu-end p-1"
                    >
                      <li>
                        <hr
                          class="dropdown-divider"
                        />
                      </li>
                      <li>
                        <a
                          class="nav-link"
                          to="#"
                        >
                          Log Out
                        </a>
                      </li>
                    </ul>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
        </div>
        <div
          class="container"
        >
          <h1>
            New Poll
          </h1>
          <form>
            <div>
              <label
                data-testid="firstOptionLabel"
              >
                First Option
              </label>
              <div
                class="mt-1"
              >
                <input
                  data-testid="firstOption"
                  id="firstOption"
                  name="firstOption"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                data-testid="secondOptionLabel"
              >
                Second Option
              </label>
              <div>
                <input
                  data-testid="secondOption"
                  id="secondOption"
                  name="secondOption"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <button
                data-testid="submit-poll"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light bg-light"
        >
          <div
            class="container-fluid"
          >
            <a
              class="nav-link"
              href="/"
            >
              <img
                alt=""
                height="30"
                loading="lazy"
                src="/logo192.png"
              />
            </a>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-bs-target="#navbarSupportedContent"
              data-bs-toggle="collapse"
              type="button"
            >
              <i
                class="fas fa-bars"
              />
            </button>
            <div
              class="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                class="navbar-nav me-auto mb-2 mb-lg-0"
              >
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/leaderboard"
                  >
                    Leaderboard
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/add"
                  >
                    New
                  </a>
                </li>
              </ul>
              <div
                data-testid="loggedUserName"
                id="loggedUserName"
              >
                Login -&gt;
              </div>
              <ul
                class="navbar-nav mb-2 mb-lg-0"
              >
                <li
                  class="nav-item dropdown"
                >
                  <a
                    aria-expanded="false"
                    class="nav-link dropdown-toggle"
                    data-bs-toggle="dropdown"
                    data-testid="navbarDropdown"
                    href="#"
                    id="navbarDropdown"
                    role="button"
                  >
                    <img
                      class="rounded-circle img-fluid"
                      height="25"
                      id="authedUserAvatar"
                      src="/login192.png"
                      width="25"
                    />
                  </a>
                  <ul
                    aria-labelledby="navbarDropdown"
                    class="dropdown-menu dropdown-menu-end p-1"
                  >
                    <li>
                      <hr
                        class="dropdown-divider"
                      />
                    </li>
                    <li>
                      <a
                        class="nav-link"
                        to="#"
                      >
                        Log Out
                      </a>
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
      <div
        class="container"
      >
        <h1>
          New Poll
        </h1>
        <form>
          <div>
            <label
              data-testid="firstOptionLabel"
            >
              First Option
            </label>
            <div
              class="mt-1"
            >
              <input
                data-testid="firstOption"
                id="firstOption"
                name="firstOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              data-testid="secondOptionLabel"
            >
              Second Option
            </label>
            <div>
              <input
                data-testid="secondOption"
                id="secondOption"
                name="secondOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <button
              data-testid="submit-poll"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
